### ISSUE TITLE
Warning being shown when closing ranger

### ISSUE THREAD

**Issue by alexanderjeurissen** [alexanderjeurissen/ranger_devicons#3]
When using devicons as the default line mode the following warning is shown when closing ranger:

```
Invalid linemode: devicons; should be metatitle/sizemtime/filename/mtime/fileinfo/permissions
```

Which is weird because the line mode can manually be activated once ranger has been invoked, and the line mode is active on activation of ranger without delay. I'll create an issue in the ranger repository inquiring about possible causes of this behaviour.

For now rest be assured that this does not impact the functionality of this plugin in any way.

**Comment by daizeng1984** [comment]
+1, I think this is happening when opening the ranger with devicon plugin. You can ctrl-z to see this error message.

**Comment by daizeng1984** [comment]
I was also kind of looking into this annoying issue. It's as @yevhen-m said in referencing issue. Here's the [commit](https://github.com/daizeng1984/ranger/commit/2d366e4f0f80fb845d940afb307281268ee94530) on my fork. Will send a PR and let's see.

**Comment by alexanderjeurissen** [comment]
@daizeng1984 I still seem to have this issue on ranger version 1.8.1 Did you change anything else to get it working ?

**Comment by daizeng1984** [comment]
@alexanderjeurissen it should be solved in new release later than 1.9.0b5 (now it's 1.9.0b6?). You might try to update to this fresh version on Github.

**Comment by alexanderjeurissen** [comment]
@daizeng1984 ok good to know, so that means it's not fixed in the stable release of Ranger (which is 1.8.1 atm afaik)

**Comment by daizeng1984** [comment]
yes, you are right.

**Comment by magnetophon** [comment]
A new ranger release is out, and it indeed does not have this issue, so it can be closed.

**Comment by alexanderjeurissen** [comment]
I just confirmed that homebrew does include the ranger release that fixes this issue (latest homebrew version is 1.9.1)

I can no longer replicate the issue in that version, and as such will be closing this issue.

**Comment by jcklpe** [comment]
Hey guys, I'm having this issue. I'm using zsh as my shell, hyper.js as my terminal shell emulator, and I'm using Ubuntu for Windows using the Windows Subsystem Layer. I installed ranger using apt install, directly from the default ubuntu install repo. Ubuntu 18.04, so Bionic Beaver. 

Loving ranger so far though!

**Comment by kronolynx** [comment]
I'm also having this problem in Ubuntu 18.04 using Urxvt both in Bash and Fish, I also installed ranger using apt.

**Comment by lbesnard** [comment]
same issue with
```
$ ranger --version
ranger version: ranger 1.9.2
Python version: 3.7.3 (default, Mar 27 2019, 22:11:17) [GCC 7.3.0]
Locale: en_AU.UTF-8
```

**Comment by aalvarado** [comment]
```
ranger
Invalid linemode: devicons; should be filename/metatitle/permissions/fileinfo/mtime/sizemtime
```

```
ranger --version
ranger version: ranger 1.9.2
Python version: 3.7.5rc1 (default, Oct  8 2019, 16:47:45) [GCC 9.2.1 20191008]
Locale: en_US.UTF-8
```

```
apt info ranger
Package: ranger
Version: 1.9.2-4ubuntu1
Priority: extra
Section: universe/utils
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Vern Sun <s5unty@gmail.com>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 853 kB
Depends: python3.7:any, python3:any, sensible-utils
Recommends: file, less, python3-chardet, w3m-img
Suggests: atool, caca-utils, sudo, poppler-utils | mupdf-tools, elinks | elinks-lite | lynx | w3m, highlight | python-pygments, unoconv, mediainfo | exiftool
Homepage: https://ranger.github.io
Download-Size: 209 kB
APT-Manual-Installed: yes
APT-Sources: http://us.archive.ubuntu.com/ubuntu eoan/universe amd64 Packages
Description: Console File Manager with VI Key Bindings
 Ranger is a console file manager with VI key bindings.  It provides a
 minimalistic and nice curses interface with a view on the directory hierarchy.
 It ships with "rifle", a file launcher that is good at automatically finding
 out which program to use for what file type.
 .
 Design Goals
  * An easily maintainable file manager in a high level language
  * A quick way to switch directories and browse the file system
  * Keep it small but useful, do one thing and do it well
  * Console based, with smooth integration into the unix shell
```

Gnome Terminal
Installed from Apt

**Comment by Pagliacii** [comment]
I encountered this problem after I installed it by the experimental method.

And I found that this plugin must installed in the plugins directory, just like `make install`.

After jump into ~/.config/ranger/plugins/ and `make install`, this warning disappeared.

**Comment by little-fish-lalala** [comment]
> @alexanderjeurissen
> 
> Hey guys, I'm having this issue. I'm using zsh as my shell, hyper.js as my terminal shell emulator, and I'm using Ubuntu for Windows using the Windows Subsystem Layer. I installed ranger using apt install, directly from the default ubuntu install repo. Ubuntu 18.04, so Bionic Beaver.
> 
> I've also installed FuraCodeMono from the precompiled nerdfont library and icons are showing up fine, no problem.
> 
> This is what my rc.conf looks like
> 
> ```
> # a plugin that adds file glyphs / icon support to Ranger:
> # https://github.com/alexanderjeurissen/ranger_devicons
> default_linemode devicons
> ```
> 
> Loving ranger so far though!
The problem disappear as I comment out the setting：
#default_linemode devicons

**Comment by wangjianze96** [comment]
Hi, I also encountered this issue and I think the reason might be ranger cannot find devicons.py's path. I installed ranger_devicon as the instruction said and no icon show up after I started ranger and with a warning "Invalid linemode: devicons; should be metatitle/sizemtime/filename/mtime/fileinfo/permissions". So I copied devicons.py and __init__.py to folder plugins/ rather than leaving them in the subfolder, and this warning is gone and all icons show up.

Hope this may help people who are still see this problem. And hope you all well!
